src/main/workspace/filtering.ts

```diff
@@ -174,13 +174,21 @@ export class BannedList {
 
       let i: number;
       for (i = 0; i < this.filters.length; i += 1) {
-        const evaluation = this.filters[i].evaluate(path);
-
-        if (this.filters[i].scope === 'FOLDER' && pathStat.isDirectory() && !evaluation) return false;
-
-        if (this.filters[i].scope === 'FILE' && pathStat.isFile() && !evaluation) return false;
-
-        if (this.filters[i].scope === 'ALL' && !evaluation) return false;
+        if (this.filters[i].scope === 'FOLDER'
```

Comment:
Why did you change the `evaluation` variable for `this.filters[i].evaluate(path)` ?
The idea here is to only evaluate once the filters